SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -717,28 +721,81 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
     const su2double beta_blended = F1_i * beta_1 + (1.0 - F1_i) * beta_2;
 
     if (dist_i > 1e-10) {
-      /*--- Production ---*/
 
       su2double diverg = 0.0;
       for (unsigned short iDim = 0; iDim < nDim; iDim++)
         diverg += PrimVar_Grad_i[iDim + idx.Velocity()][iDim];
 
       /*--- If using UQ methodolgy, calculate production using perturbed Reynolds stress matrix ---*/
 
+      const su2double VorticityMag = GeometryToolbox::Norm(3, Vorticity_i);
       su2double StrainMag = StrainMag_i;
+      su2double P_Base = StrainMag;  //Base production term for SST1994 and SST2003
 
-      if (using_uq) {
-        ComputePerturbedRSM(nDim, Eig_Val_Comp, uq_permute, uq_delta_b, uq_urlx, PrimVar_Grad_i + idx.Velocity(),
+      /*--- Apply production term modifications ---*/
+      switch (sstParsedOptions.production) {
+        case SST_OPTIONS::UQ:
+          ComputePerturbedRSM(nDim, Eig_Val_Comp, uq_permute, uq_delta_b, uq_urlx, PrimVar_Grad_i + idx.Velocity(),
                             Density_i, Eddy_Viscosity_i, ScalarVar_i[0], MeanPerturbedRSM);
-        StrainMag = PerturbedStrainMag(ScalarVar_i[0]);
+          StrainMag = PerturbedStrainMag(ScalarVar_i[0]);
+          P_Base = PerturbedStrainMag(ScalarVar_i[0]);
+          break;
+
+        case SST_OPTIONS::V:
+          P_Base = VorticityMag;
+          break;
+
+        case SST_OPTIONS::KL:
+          P_Base = sqrt(StrainMag*VorticityMag);
+          break;
+
+        default:
+          P_Base = StrainMag;
       }
 
-      su2double pk = Eddy_Viscosity_i * pow(StrainMag, 2) - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
-      pk = max(0.0, min(pk, 20.0 * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]));
+      su2double pk = Eddy_Viscosity_i * pow(P_Base, 2);
 
-      const su2double VorticityMag = GeometryToolbox::Norm(3, Vorticity_i);
-      const su2double zeta = max(ScalarVar_i[1], VorticityMag * F2_i / a1);
-      su2double pw = alfa_blended * Density_i * max(pow(StrainMag, 2) - 2.0 / 3.0 * zeta * diverg, 0.0);
+   //   if (sstParsedOptions.version != SST_OPTIONS::MODIFIED) {
+       /* in case of unmodified, we add the divergence terms */
+        if (sstParsedOptions.version == SST_OPTIONS::V1994) {
+          /*--- INTRODUCE THE SST-V1994 BUG WHERE DIVERGENCE TERM IS MISSING ---*/
+          /*--- Note that we only keep this divergence term for the 1994v and 2003v model (vorticity)---*/
+          pk = pk - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
+        } 
+    //  } 
+
+      /*--- production limiter ---*/
+      pk = min(pk, ProdLimConstant * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]);
+      pk = max(0.0, pk);
+
+      su2double zeta;
+
+      if (sstParsedOptions.version == SST_OPTIONS::V1994){
+        zeta = max(ScalarVar_i[1], VorticityMag * F2_i / a1);
+        /*--- ---*/
+      } else {
+        zeta = max(ScalarVar_i[1], StrainMag_i * F2_i / a1);
+      }
+
+      su2double pw = pow(StrainMag, 2); 
```

Comment:
I think this is wrong, the term used in the production is the same for both equations.

Reply:
Yes, production should contain strain in the zeta term for both models, this is another difference between the literature V1994m and the su2 implementation. su2 mixed with the vorticity model here.